File: Backend/tournamentservice/src/main/java/com/PlayProFootball/tournamentservice/entity/Tournament.java
package com.PlayProFootball.tournamentservice.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "tournament")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Tournament {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "state_id", nullable = false)
    private State state;

    private String city;

    private LocalDate startDate;
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    private TournamentType type;

    @Column(name = "ispublished")
    private boolean isPublished;

}

File: Backend/TouramentDetails/Models/Tournament.cs
using System;
using System.Collections.Generic;

namespace TournamentDetails.Models
{
    public class Tournament
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int StateId { get; set; }
        public string City { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string Type { get; set; }
        public int? Oid { get; set; }
        public bool IsPublished { get; set; }
        public DateTime? RegistrationDeadline { get; set; }

        public ICollection<TournamentTeam> TournamentTeams { get; set; }
        public ICollection<Match> Matches { get; set; }
    }
}